packages/task/src/browser/task-service.ts

```diff
@@ -1008,10 +1015,14 @@ export class TaskService implements TaskConfigurationClient {
                 taskConfig: taskInfo ? taskInfo.config : undefined
             }
         );
-        widget.start(processId);
+        widget.start(terminalId);
     }
 
     private getTerminalWidgetId(terminalId: number): string {
+        const terminalWidget = this.terminalService.getByTerminalId(terminalId);
+        if (terminalWidget) {
+            return terminalWidget.id;
+        }
         return `${TERMINAL_WIDGET_FACTORY_ID}-${terminalId}`;
```

Comment:
it looks like broken abstraction, only the terminal widget factory should know how ids are created, clients should not assume it, otherwise they will be broken if the factory is changed. Should not we return `undefined` rather here? And remove references to `TERMINAL_WIDGET_FACTORY_ID`.

Reply:
You are right. I removed the line that has the factory id, and re-tested.